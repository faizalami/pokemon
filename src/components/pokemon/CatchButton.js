/** @jsxImportSource @emotion/react */
import { Button } from '../Buttons';
import Image from '../Image';
import { css, keyframes } from '@emotion/react';
import { margin, padding, rounded } from '../utilities';

const shake = keyframes`
  0% {
    transform: translate(1px, 1px) rotate(0deg);
  }
  10% {
    transform: translate(-1px, -2px) rotate(-1deg);
  }
  20% {
    transform: translate(-3px, 0px) rotate(1deg);
  }
  30% {
    transform: translate(3px, 2px) rotate(0deg);
  }
  40% {
    transform: translate(1px, -1px) rotate(1deg);
  }
  50% {
    transform: translate(-1px, 2px) rotate(-1deg);
  }
  60% {
    transform: translate(-3px, 1px) rotate(0deg);
  }
  70% {
    transform: translate(3px, 1px) rotate(-1deg);
  }
  80% {
    transform: translate(-1px, -1px) rotate(1deg);
  }
  90% {
    transform: translate(1px, 2px) rotate(0deg);
  }
  100% {
    transform: translate(1px, -2px) rotate(-1deg);
  }
`;

const iconShake = css`
  & img:hover,
  &:focus img {
    animation: ${shake} 0.5s;
    animation-iteration-count: infinite;
  }
`;

const buttonLabel = css`
  background-color: white;
  font-weight: bold;
  ${rounded}
  ${padding.a1}
  ${margin.a0}
`;

function CatchButton ({ className, onClick }) {
  return (
    <Button column variant="link" onClick={onClick} className={className} css={iconShake}>
      <Image
        png="/assets/img/png/pokeball-64.png"
        webp="/assets/img/webp/pokeball-64.webp"
        alt="Catch!"
        lazy
        width={64}
        height={64}
      />
      <p css={buttonLabel}>Catch!</p>
    </Button>
  );
}

export default CatchButton;
