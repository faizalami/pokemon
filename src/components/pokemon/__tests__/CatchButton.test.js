import { render } from '@testing-library/react';
import CatchButton from '../CatchButton';

test('Catch Button matches snapshot', () => {
  expect(render(<CatchButton/>)).toMatchSnapshot();
});
