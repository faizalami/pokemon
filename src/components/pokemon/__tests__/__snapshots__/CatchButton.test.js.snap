// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catch Button matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": @keyframes animation-0 {
  0% {
    -webkit-transform: translate(1px, 1px) rotate(0deg);
    -moz-transform: translate(1px, 1px) rotate(0deg);
    -ms-transform: translate(1px, 1px) rotate(0deg);
    transform: translate(1px, 1px) rotate(0deg);
  }

  10% {
    -webkit-transform: translate(-1px, -2px) rotate(-1deg);
    -moz-transform: translate(-1px, -2px) rotate(-1deg);
    -ms-transform: translate(-1px, -2px) rotate(-1deg);
    transform: translate(-1px, -2px) rotate(-1deg);
  }

  20% {
    -webkit-transform: translate(-3px, 0px) rotate(1deg);
    -moz-transform: translate(-3px, 0px) rotate(1deg);
    -ms-transform: translate(-3px, 0px) rotate(1deg);
    transform: translate(-3px, 0px) rotate(1deg);
  }

  30% {
    -webkit-transform: translate(3px, 2px) rotate(0deg);
    -moz-transform: translate(3px, 2px) rotate(0deg);
    -ms-transform: translate(3px, 2px) rotate(0deg);
    transform: translate(3px, 2px) rotate(0deg);
  }

  40% {
    -webkit-transform: translate(1px, -1px) rotate(1deg);
    -moz-transform: translate(1px, -1px) rotate(1deg);
    -ms-transform: translate(1px, -1px) rotate(1deg);
    transform: translate(1px, -1px) rotate(1deg);
  }

  50% {
    -webkit-transform: translate(-1px, 2px) rotate(-1deg);
    -moz-transform: translate(-1px, 2px) rotate(-1deg);
    -ms-transform: translate(-1px, 2px) rotate(-1deg);
    transform: translate(-1px, 2px) rotate(-1deg);
  }

  60% {
    -webkit-transform: translate(-3px, 1px) rotate(0deg);
    -moz-transform: translate(-3px, 1px) rotate(0deg);
    -ms-transform: translate(-3px, 1px) rotate(0deg);
    transform: translate(-3px, 1px) rotate(0deg);
  }

  70% {
    -webkit-transform: translate(3px, 1px) rotate(-1deg);
    -moz-transform: translate(3px, 1px) rotate(-1deg);
    -ms-transform: translate(3px, 1px) rotate(-1deg);
    transform: translate(3px, 1px) rotate(-1deg);
  }

  80% {
    -webkit-transform: translate(-1px, -1px) rotate(1deg);
    -moz-transform: translate(-1px, -1px) rotate(1deg);
    -ms-transform: translate(-1px, -1px) rotate(1deg);
    transform: translate(-1px, -1px) rotate(1deg);
  }

  90% {
    -webkit-transform: translate(1px, 2px) rotate(0deg);
    -moz-transform: translate(1px, 2px) rotate(0deg);
    -ms-transform: translate(1px, 2px) rotate(0deg);
    transform: translate(1px, 2px) rotate(0deg);
  }

  100% {
    -webkit-transform: translate(1px, -2px) rotate(-1deg);
    -moz-transform: translate(1px, -2px) rotate(-1deg);
    -ms-transform: translate(1px, -2px) rotate(-1deg);
    transform: translate(1px, -2px) rotate(-1deg);
  }
}

.emotion-0 {
  border-radius: 0.375rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1rem;
  cursor: pointer;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  background-color: transparent;
  border: 1px solid transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  object-fit: contain;
  object-position: center;
}

.emotion-1:hover {
  -webkit-animation: animation-0 0.5s;
  animation: animation-0 0.5s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
}

.emotion-2 {
  background-color: white;
  font-weight: bold;
  border-radius: 0.5rem;
  padding: 0.25rem;
  margin: 0rem;
}

<body>
    <div>
      <button
        class="emotion-0"
      >
        <picture>
          <source
            data-testid="image-webp"
            srcset="/assets/img/webp/pokeball-64.webp"
            type="image/webp"
          />
          <source
            data-testid="image-png"
            srcset="/assets/img/png/pokeball-64.png"
            type="image/png"
          />
          <img
            alt="Catch!"
            class="emotion-1"
            data-testid="image-regular"
            height="64"
            loading="lazy"
            src="/assets/img/png/pokeball-64.png"
            width="64"
          />
        </picture>
        <p
          class="emotion-2"
        >
          Catch!
        </p>
      </button>
    </div>
  </body>,
  "container": @keyframes animation-0 {
  0% {
    -webkit-transform: translate(1px, 1px) rotate(0deg);
    -moz-transform: translate(1px, 1px) rotate(0deg);
    -ms-transform: translate(1px, 1px) rotate(0deg);
    transform: translate(1px, 1px) rotate(0deg);
  }

  10% {
    -webkit-transform: translate(-1px, -2px) rotate(-1deg);
    -moz-transform: translate(-1px, -2px) rotate(-1deg);
    -ms-transform: translate(-1px, -2px) rotate(-1deg);
    transform: translate(-1px, -2px) rotate(-1deg);
  }

  20% {
    -webkit-transform: translate(-3px, 0px) rotate(1deg);
    -moz-transform: translate(-3px, 0px) rotate(1deg);
    -ms-transform: translate(-3px, 0px) rotate(1deg);
    transform: translate(-3px, 0px) rotate(1deg);
  }

  30% {
    -webkit-transform: translate(3px, 2px) rotate(0deg);
    -moz-transform: translate(3px, 2px) rotate(0deg);
    -ms-transform: translate(3px, 2px) rotate(0deg);
    transform: translate(3px, 2px) rotate(0deg);
  }

  40% {
    -webkit-transform: translate(1px, -1px) rotate(1deg);
    -moz-transform: translate(1px, -1px) rotate(1deg);
    -ms-transform: translate(1px, -1px) rotate(1deg);
    transform: translate(1px, -1px) rotate(1deg);
  }

  50% {
    -webkit-transform: translate(-1px, 2px) rotate(-1deg);
    -moz-transform: translate(-1px, 2px) rotate(-1deg);
    -ms-transform: translate(-1px, 2px) rotate(-1deg);
    transform: translate(-1px, 2px) rotate(-1deg);
  }

  60% {
    -webkit-transform: translate(-3px, 1px) rotate(0deg);
    -moz-transform: translate(-3px, 1px) rotate(0deg);
    -ms-transform: translate(-3px, 1px) rotate(0deg);
    transform: translate(-3px, 1px) rotate(0deg);
  }

  70% {
    -webkit-transform: translate(3px, 1px) rotate(-1deg);
    -moz-transform: translate(3px, 1px) rotate(-1deg);
    -ms-transform: translate(3px, 1px) rotate(-1deg);
    transform: translate(3px, 1px) rotate(-1deg);
  }

  80% {
    -webkit-transform: translate(-1px, -1px) rotate(1deg);
    -moz-transform: translate(-1px, -1px) rotate(1deg);
    -ms-transform: translate(-1px, -1px) rotate(1deg);
    transform: translate(-1px, -1px) rotate(1deg);
  }

  90% {
    -webkit-transform: translate(1px, 2px) rotate(0deg);
    -moz-transform: translate(1px, 2px) rotate(0deg);
    -ms-transform: translate(1px, 2px) rotate(0deg);
    transform: translate(1px, 2px) rotate(0deg);
  }

  100% {
    -webkit-transform: translate(1px, -2px) rotate(-1deg);
    -moz-transform: translate(1px, -2px) rotate(-1deg);
    -ms-transform: translate(1px, -2px) rotate(-1deg);
    transform: translate(1px, -2px) rotate(-1deg);
  }
}

.emotion-0 {
  border-radius: 0.375rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1rem;
  cursor: pointer;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  background-color: transparent;
  border: 1px solid transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  object-fit: contain;
  object-position: center;
}

.emotion-1:hover {
  -webkit-animation: animation-0 0.5s;
  animation: animation-0 0.5s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
}

.emotion-2 {
  background-color: white;
  font-weight: bold;
  border-radius: 0.5rem;
  padding: 0.25rem;
  margin: 0rem;
}

<div>
    <button
      class="emotion-0"
    >
      <picture>
        <source
          data-testid="image-webp"
          srcset="/assets/img/webp/pokeball-64.webp"
          type="image/webp"
        />
        <source
          data-testid="image-png"
          srcset="/assets/img/png/pokeball-64.png"
          type="image/png"
        />
        <img
          alt="Catch!"
          class="emotion-1"
          data-testid="image-regular"
          height="64"
          loading="lazy"
          src="/assets/img/png/pokeball-64.png"
          width="64"
        />
      </picture>
      <p
        class="emotion-2"
      >
        Catch!
      </p>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
